### PR TITLE
HypervisorManagerPR

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
Added support to HypervisorManager for API 800/1000/1200

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [x] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

**Review by soodpr — CHANGES_REQUESTED** [review]
Unit test cases are missing

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
 ## 3.2.1(unreleased)
+### Notes
+- This release adds support to Oneview Rest API version 800, 1000, and 1200 for C7000 and Synergy to the features Hypervisor Manager.
 
-## Bugs & Enhancements:
-- [#419](https://github.com/HewlettPackard/oneview-chef/issues/419) Dockerfile is failing because of insufficient libraries.
```
Comment (soodpr) [inline CHANGELOG.md:9]: we don't have to remove anything which is not yet released

Comment (AsisBagga) [reply CHANGELOG.md:9]: @soodpr Corrected.

File: examples/hypervisor_manager.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+# Note: This resources will work for client version greater than API800.
+
+my_client = {
+  url: ENV['ONEVIEWSDK_URL'],
+  user: ENV['ONEVIEWSDK_USER'],
+  password: ENV['ONEVIEWSDK_PASSWORD'],
+  api_version: 800
```
Comment (AnushaY1916) [inline examples/hypervisor_manager.rb:17]: Keep the latest version.

**Review by AnushaY1916 — APPROVED** [review]

File: examples/hypervisor_manager.rb (hunk 2 above)
Comment (AsisBagga) [reply examples/hypervisor_manager.rb:17]: @AnushaY1916 Corrected

File: libraries/matchers.rb [hunk 3]
```diff
@@ -31,6 +31,7 @@
     oneview_fc_network:                 standard_actions + scope_actions + %i[reset_connection_template],
     oneview_fcoe_network:               standard_actions + scope_actions + %i[reset_connection_template],
     oneview_firmware:                   %i[add remove create_custom_spp],
+    oneview_hypervisor_manager:         standard_actions + %i[update_registration create_from_update],
```
Comment (soodpr) [inline libraries/matchers.rb]: remove create_from_update action from here

File: libraries/resource_providers/api800/c7000/hypervisor_manager_provider.rb [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            if @item.like? desired_state
+              Chef::Log.info("#{@resource_name} '#{@name}' is up to date")
+            else
+              diff = get_diff(@item, desired_state)
+              Chef::Log.info "#{method2.to_s.capitalize} #{@resource_name} '#{@name}'#{diff}"
+              @context.converge_by "#{method2.to_s.capitalize} #{@resource_name} '#{@name}'" do
+                @item.update(desired_state)
+              end
+            end
+          else
```
Comment (soodpr) [inline libraries/resource_providers/api800/c7000/hypervisor_manager_provider.rb:33]: As discussed, we don't have to create the resource if the request is to update the registration.

File: libraries/matchers.rb (hunk 3 above)
Comment (AsisBagga) [reply libraries/matchers.rb]: @soodpr Corrected

File: libraries/resource_providers/api800/c7000/hypervisor_manager_provider.rb (hunk 4 above)
Comment (AsisBagga) [reply libraries/resource_providers/api800/c7000/hypervisor_manager_provider.rb:33]: @soodpr Corrected

File: libraries/matchers.rb [hunk 5]
```diff
@@ -31,7 +31,11 @@
     oneview_fc_network:                 standard_actions + scope_actions + %i[reset_connection_template],
     oneview_fcoe_network:               standard_actions + scope_actions + %i[reset_connection_template],
     oneview_firmware:                   %i[add remove create_custom_spp],
+<<<<<<< HEAD
```
Comment (soodpr) [inline libraries/matchers.rb]: please remove these rebase related commit lines..

File: libraries/matchers.rb [hunk 6]
```diff
@@ -31,7 +31,11 @@
...
+    oneview_hypervisor_manager:         standard_actions + %i[update_registration],
+=======
```
Comment (soodpr) [inline libraries/matchers.rb]: remove this as well

File: libraries/matchers.rb [hunk 7]
```diff
@@ -31,7 +31,11 @@
...
     oneview_hypervisor_cluster_profile: standard_actions + %i[update],
+>>>>>>> 087b5c24702759c6616f5eb19b3726c25499d8d4
```
Comment (soodpr) [inline libraries/matchers.rb]: Remove this line as well

File: libraries/matchers.rb (hunk 5 above)
Comment (AsisBagga) [reply libraries/matchers.rb]: @soodpr Corrected

File: libraries/matchers.rb (hunk 6 above)
Comment (AsisBagga) [reply libraries/matchers.rb]: @soodpr Corrected

File: libraries/matchers.rb (hunk 7 above)
Comment (AsisBagga) [reply libraries/matchers.rb]: @soodpr Corrected

**Review by soodpr — APPROVED** [review]

**Pull request merged**